src/Map.jl

```diff
@@ -21,6 +21,33 @@ function check_composable(a::Map, b::Map)
    codomain(a) != domain(b) && error("Incompatible maps")
 end
 
+function Base.show(io::IO, ::MIME"text/plain", M::AbstractAlgebra.Map)
+   # the "header" is identical to the supercompact output; this
+   # allows other map types to reuse this method
+   println(IOContext(io, :supercompact => true), M)
+   io = pretty(io)
+   println(io, Indent(), "from ", Lowercase(), domain(M))
+   print(io, "to ", Lowercase(), codomain(M), Dedent())
+   show_map_data(io, M)
+end
```

Comment:
This one method replaces the "detailed" show methods for all `Maps` subtypes in AA. Note how it (ab) uses supercompact printing to get the lead part, and calls out to a helper `show_map_data` to allow showing extra data.